### PR TITLE
Check auth before proceeding such that we don't show a badge if auth fails

### PR DESCRIPTION
**Pull request by SethNute**

### CODE REVIEW

**Comment by sppatel** [comment]
I like the intent here.. if we are going to fail, fail early.  Though I think it would be even one step better to validate the creds when you configure the sauce labs service instance.  Is there anything we can do in the toolbroker?

**Comment by eLobeto** [comment]
Sauce.py doesn't get run until after the tests try to get executed so this auth check would need to be moved to extension.json with cURL

**Comment by SethNute** [comment]
@sppatel I know that if you try to use faulty access credentials when manually editing the saucelabs tool card it will reject them. However, you can still invalidate your access token (after its set in the toolchain) via the saucelabs API/UI and I don't think theres anything the toolbroker can do to handle that. Also, you can use invalidate credentials when creating a toolchain from a template, but I've requested we do some checking there.

File: extension.json [hunk 1]
```diff
@@ -107,6 +107,14 @@ set +x
 set +e 
 export INIT_START_TIME=$(date +\"%s\")
 
+#############
+# Colors    #
+#############
+export green=\'\e[0\;32m\'
+export red=\'\e[0\;31m\'
+export label_color=\'\e[0\;33m\'
+export no_color=\'\e[0m\'
```
Comment (SethNute) [inline extension.json]: I _think_ all of this escaping should work properly when actually run in the context of the pipeline, @eLobeto is there a way I could test this branch out on dev?

**Comment by SethNute** [comment]
@eLobeto This is looking good now, will merging this break the YP extension?

**Comment by eLobeto** [comment]
@SethNute if only extension.json is updated it won't effect the existing extension. If one of the other scripts changes then every time its cloned it will get the new stuff.. so you should be good.

**Pull request merged**